### PR TITLE
Advanced sys def update

### PR DESCRIPTION
**Pull request by DStavilaNI**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-custom-device-development-tools/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- Adds the latest updates for the Advanced System API, from the original repository to the corresponding subtree of the NI CD Development Tools.

### Why should this Pull Request be merged?

- Implemented CDS structures with OS === Win and OS === RT(Default) options for the commited VIs:

- advanced-sysdef-api/Source/_SubVIs/SysDef.Node.Info.Get.vi
- advanced-sysdef-api/Source/_SubVIs/TreeItem.Get.vi
- advanced-sysdef-api/Source/_SubVIs/TreeItem.Property.Get.vi
- advanced-sysdef-api/Source/_SubVIs/TreeItem.Property.Set.vi
- advanced-sysdef-api/Source/_SubVIs/_TreeItem.Property.Get.Dictionary.Info.vi
- advanced-sysdef-api/Source/_SubVIs/_TreeItem.Property.Get.Dictionary.Read.vi
- advanced-sysdef-api/Source/_SubVIs/_TreeItem.Property.Get.Dictionary.vi
- advanced-sysdef-api/Source/_SubVIs/_TreeItem.Property.Set.Dictionary.Info.vi
- advanced-sysdef-api/Source/_SubVIs/_TreeItem.Property.Set.Dictionary.Write.vi
- advanced-sysdef-api/Source/_SubVIs/_TreeItem.Property.Set.Dictionary.vi

The lack of CDS structures with these options was causing compile errors when building Engine libraries for Custom Devices for RT Targets. For some reason, due to cross-referencing, the Advanced System Definition API library and some of it's member VIs were included in the Engine builds for the RT targets, and this was leading to errors during builds, since .NET controls, used in this library are not supported on RT Targets.

### What testing has been done?

- Local manual testing has been done.

### CODE REVIEW

**Comment by rtzoeller** [comment]
A few administrative items before I dive into the code:

> Fixes a bunch of bugs, related to the CDS and the used symbols in the updated VIs (the complete details are in the descriptions of the added commits from the original repo).

- [ ] I don't want to rely on the other repository continuing to exist (it will likely eventually be deleted). Please update this PR to stand alone.

> LV 2016, as in the original repository is used for these VIs.

- [ ] c4b7d866a1dfe5f87844439f58b75877755dd669 mass-compiled this project to LabVIEW 2017. Your change should also be in LabVIEW 2017, as listed in the repository [README](https://github.com/ni/niveristand-custom-device-development-tools#labview-version).

> Without these fixes, the referred API will simply not work if used in the context of the VeriStand System Explorer.

- [ ] Do you mean outside of System Explorer? It seems like the original code should work within System Explorer just fine. Please elaborate on the problem you are encountering with the existing code further (e.g. reproducing steps).

File: advanced-sysdef-api/README.md [hunk 1]
```diff
@@ -3,3 +3,30 @@ VeriStand Addon System Definition Utilities
...
+### Usage ###
+Install the VI package in the Built folder.
+
+### LabVIEW Version ###
+
+LabVIEW 2016
+
+### Built Availability ###
+
+Use the attached VI Package in the Built folder.
```
Comment (rtzoeller) [inline advanced-sysdef-api/README.md:15]: Please revert the changes to this file.

Comment (DStavilaNI) [reply advanced-sysdef-api/README.md]: done

**Comment by DStavilaNI** [comment]
> 
> 
> A few administrative items before I dive into the code:
> 
> > Fixes a bunch of bugs, related to the CDS and the used symbols in the updated VIs (the complete details are in the descriptions of the added commits from the original repo).
> 
>     * [ ]  I don't want to rely on the other repository continuing to exist (it will likely eventually be deleted). Please update this PR to stand alone.
> 
> 
> > LV 2016, as in the original repository is used for these VIs.
> 
>     * [ ]  [c4b7d86](https://github.com/ni/niveristand-custom-device-development-tools/commit/c4b7d866a1dfe5f87844439f58b75877755dd669) mass-compiled this project to LabVIEW 2017. Your change should also be in LabVIEW 2017, as listed in the repository [README](https://github.com/ni/niveristand-custom-device-development-tools#labview-version).
> 
> 
> > Without these fixes, the referred API will simply not work if used in the context of the VeriStand System Explorer.
> 
>     * [ ]  Do you mean outside of System Explorer? It seems like the original code should work within System Explorer just fine. Please elaborate on the problem you are encountering with the existing code further (e.g. reproducing steps).
> 
> 
> A few administrative items before I dive into the code:
> 
> > Fixes a bunch of bugs, related to the CDS and the used symbols in the updated VIs (the complete details are in the descriptions of the added commits from the original repo).
> 
>     * [ ]  I don't want to rely on the other repository continuing to exist (it will likely eventually be deleted). Please update this PR to stand alone.
> 
> 
> > LV 2016, as in the original repository is used for these VIs.
> 
>     * [ ]  [c4b7d86](https://github.com/ni/niveristand-custom-device-development-tools/commit/c4b7d866a1dfe5f87844439f58b75877755dd669) mass-compiled this project to LabVIEW 2017. Your change should also be in LabVIEW 2017, as listed in the repository [README](https://github.com/ni/niveristand-custom-device-development-tools#labview-version).
> 
> 
> > Without these fixes, the referred API will simply not work if used in the context of the VeriStand System Explorer.
> 
>     * [ ]  Do you mean outside of System Explorer? It seems like the original code should work within System Explorer just fine. Please elaborate on the problem you are encountering with the existing code further (e.g. reproducing steps).

- Comments 1,2 are addressed.
- Comment 3: No, I meant within the System Explorer. First of all, the Advanced System Definition API code within monorepo is not up-to-date with the Advanced System Definition API code within the original repo. 

       As it is in the monorepo, if for some reason, one or more of the committed/modified VIs gets included as a dependency in a RT build, it will break that build because of the .NET nodes. This was a problem for us with the IA builds. 

      Therefore, after some discussions with Daniel E., we decided to implement CDS structures for the above mentioned code. The initial implementation was with "OFFLINE==TRUE" and "OFFLINE==FALSE" keys, however that led to a second problem -  that code was not being executed anymore in the VeriStand System Explorer context.  Consider for example this snippet in the initial implementation:

Example Image

OFFLINE==TRUE - means that the code can be executed in a LV Project containing the OFFLINE=TRUE symbol. 
OFFLINE == FALSE - and this is an empty case. This means that nothing will be executed while in VeriStand context. 

So we replaced these keys with with "OS == Win" and "OS == RT(Default)" options for the committed VIs in a second implementation, as they reflect better the operation mode that is expected from this API

I hope this helps.

**Comment by rtzoeller** [comment]
It seems like a number of VIs were saved in your last commit, but without any changes. Please revert these back to the version on `main`.

**Comment by DStavilaNI** [comment]
I will resubmit this PR, I think it would be simpler
